de.dlr.sc.virsat.model.test/src/de/dlr/sc/virsat/model/dvlm/tree/TreeTraverserTest.java

```diff
@@ -0,0 +1,169 @@
+		seiChild1Child.setType(se);
+		seiChild1ChildChild.setType(se);
+		seiChild2.setType(se);
+		seiChild2Child.setType(se);
+		seiChild2ChildChild.setType(se);
+		
+		seiRoot.getChildren().add(seiChild1);
+		seiRoot.getChildren().add(seiChild2);
+		seiChild1.getChildren().add(seiChild1Child);
+		seiChild2.getChildren().add(seiChild2Child);
+		seiChild1Child.getChildren().add(seiChild1ChildChild);
+		seiChild2Child.getChildren().add(seiChild2ChildChild);
+		
+		// Now set names for the 2 level and for the 3 level,
+		// Try to find them and check if the traverser reacts as expected
+		seiRoot.setName("Root");
+		seiChild1Child.setName("ChildOf1");
+		seiChild2Child.setName("ChildOf2");
+		seiChild1ChildChild.setName("DoubleChildOf1");
+		seiChild2ChildChild.setName("DoubleChildOf2");
+	}
+
+	/**
+	 * Class to capture the results from the tree traversing
+	 */
+	class Result {
+		
+		@Override
+		public int hashCode() {
+			final int prime = 31;
+			int result = 1;
+			result = prime * result + Objects.hash(matchingLevel, nestingLevel, sei);
+			return result;
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			} else if (obj == null) {
+				return false;
+			} else if (getClass() != obj.getClass()) {
+				return false;
+			} else {
+				Result other = (Result) obj;
+				return matchingLevel == other.matchingLevel && nestingLevel == other.nestingLevel && Objects.equals(sei, other.sei);
+			}
+		}
+		
+		Result(StructuralElementInstance sei, int processedLevel, int matchedLevel) {
+			this.sei = sei;
+			this.nestingLevel = processedLevel;
+			this.matchingLevel = matchedLevel;
+		}
+
+		StructuralElementInstance sei;
+		int nestingLevel;
+		int matchingLevel;
+	}
+	
+	@Test
+	public void testTraverse() {
+		List<Result> traverseResults = new LinkedList<>();
+		Map<StructuralElementInstance, StructuralElementInstance> mapToParentResults = new HashMap<>();
+		
+		TreeTraverser<StructuralElementInstance> traverser = new TreeTraverser<>();
+		
+		traverser.traverse(seiRoot, new IStructuralElementInstanceTreeTraverserMatcher() {
+			@Override
+			public void processMatch(StructuralElementInstance treeNode, StructuralElementInstance matchingParent) {
+				mapToParentResults.put(treeNode, matchingParent);
+			}
+			
+			@Override
+			public boolean isMatching(StructuralElementInstance treeNode) {
+				return treeNode.getName() != null;
+			}
+			
+			@Override
+			public boolean continueTraverseChildren(StructuralElementInstance treeNode, boolean isMatching, int nestingLevel, int matchingLevel) {
+				traverseResults.add(new Result(treeNode, nestingLevel, matchingLevel));
+				
+				// Try to make sure that the seiChild2ChildChild which is theoretically correct, still does not get found
+				boolean isChild2Child = treeNode == seiChild2Child;
+				return !isChild2Child && IStructuralElementInstanceTreeTraverserMatcher.super.continueTraverseChildren(treeNode, isMatching, nestingLevel, matchingLevel);

```

Comment:
```suggestion
				return treeNode != seiChild2Child;

```

Reply:
The former one, has the default implementation of the interface connected to the test. Thus a change to the default implementation would be captured here. I prefer leaving it as it is, but i added a comment, why the statement has to be that complex.